src/components/tables/EditableTranscriptsTable.tsx

```diff
@@ -0,0 +1,227 @@
+import axios from "axios";
+import { signIn, signOut, useSession } from "next-auth/react";
+import { useRouter } from "next/router";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import { Transcript } from "../../../types";
+import BaseTable from "./BaseTable";
+import { TableStructure } from "./types";
+
+const EmptyText = ({
+  hasActiveTranscript,
+  hasMorePendingTranscript,
+}: {
+  hasMorePendingTranscript: boolean;
+  hasActiveTranscript: boolean;
+}) => {
+  const props =
+    hasMorePendingTranscript || hasActiveTranscript
+      ? {
+          color: "black",
+          fontWeight: "bold",
+        }
+      : undefined;
+  const text = useMemo(() => {
+    if (hasActiveTranscript)
+      return "Please finish editing & submit the transcript you're working on before choosing a new one";
+    if (hasMorePendingTranscript)
+      return "You can only have 3 transcripts under review at once. Please wait until at least one is approved";
+
+    return "There are no transcripts that you can edit";
+  }, [hasActiveTranscript, hasMorePendingTranscript]);
+
+  return <Text {...props}>{text}</Text>;
+};
+
+const activeLimit = 1;
+const pendingLimit = 3;
+
+const EditableTranscriptsTable = () => {
+  const { data: session, status } = useSession();
+  const { data: activeReviews = [] } = useUserReviews({
+    userId: session?.user?.id,
+    status: "active",
+  });
+  const { data: pendingReviews = [] } = useUserReviews({
+    userId: session?.user?.id,
+    status: "pending",
+  });
+
+  const router = useRouter();
+  const claimTranscript = useClaimTranscript();
+  const { data, isLoading, isError, refetch } = useTranscripts();
+  const toast = useToast();
+  const hasActiveTranscript = activeReviews.length >= activeLimit;
+  const hasMorePendingTranscript = pendingReviews.length >= pendingLimit;
+  const editableData = useMemo(() => {
+    if (hasActiveTranscript || hasMorePendingTranscript) return [];
+
+    return data;
+  }, [data, hasActiveTranscript, hasMorePendingTranscript]);
+
+  const retriedClaim = useRef(0);
+
+  const [claimState, setClaimState] = useState({
+    rowId: -1,
+  });
+
+  const retryLoginAndClaim = async (transcriptId: number) => {
+    await signOut({ redirect: false });
+    if (retriedClaim.current < 2) {
+      retriedClaim.current += 1;
+      await signIn("github", {
+        callbackUrl: `/?reclaim=true&txId=${transcriptId}`,
+      });
+    }
+  };
+
+  const handleClaim = useCallback(
+    async (transcriptId: number) => {
+      if (status === "loading") {
+        toast({
+          status: "loading",
+          title: "loading fun",
+          description: "loading up some fun for you... please wait.",
+        });
+        return;
+      } else if (status === "unauthenticated") {
+        // Sign-in user before trying to claim a transcript
+        await signIn("github", {
+          callbackUrl: `/?reclaim=true&txId=${transcriptId}`,
```

Comment:
txId looks more suited for transaction id and this is for transcript id. Consider using tscId or transcriptId itself